File: dotNet/dotNetWrapper/ProcessingSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Abbyy.CloudOcrSdk
{
    public enum OutputFormat
    {
        txt, rtf, docx, xlsx, pptx, pdfSearchable, pdfTextAndImages, xml
    };

    public enum RecognitionLanguage
    {
        English, French, Italian, German, Spanish, Russian, ChinesePRC, Japanese, Korean
    };

    public class ProcessingSettings : IProcessingSettings
    {
        public ProcessingSettings()
        {
            Language = RecognitionLanguage.English;
        }

        public RecognitionLanguage Language
        {
            set { _language = value.ToString(); }
        }

        public void SetLanguage(string lang)
        {
            _language = lang;
        }

        public OutputFormat OutputFormat { get; set; }

        public string Description { get; set; }

        public TextType TextTypes
        {
            get;
            set;
        }

        /// <summary>
        /// Any Url parameters that are passed as-is to the server
        /// </summary>
        public String CustomOptions
        {
            get;
            set;
        }

        /// <summary>
        /// Gets default extension for given output format
        /// </summary>
        public string OutputFileExt
        {
            get
            {
                switch (this.OutputFormat)
                {
                    case OutputFormat.docx:
                        return ".docx";
                    case OutputFormat.pdfSearchable:
                    case OutputFormat.pdfTextAndImages:
                        return ".pdf";
                    case OutputFormat.pptx:
                        return ".pptx";
                    case OutputFormat.rtf:
                        return ".rtf";
                    case OutputFormat.txt:
                        return ".txt";
                    case OutputFormat.xlsx:
                        return ".xlsx";
                    case OutputFormat.xml:
                        return ".xml";
                }

                return ".bin";
            }
        }

        public string AsUrlParams
        {
            get
            {
                StringBuilder result = new StringBuilder();
                result.Append(String.Format("language={0}&exportFormat={1}",
                    Uri.EscapeDataString(_language),
                    Uri.EscapeDataString(OutputFormat.ToString())));

                string textType = TextTypes.AsUrlParams();
                if (!String.IsNullOrEmpty(textType))
                    result.AppendFormat("&textType={0}", textType);

                if (!String.IsNullOrEmpty(CustomOptions))
                    result.Append("&"+CustomOptions);

                return result.ToString();
            }
        }

        private string _language = "english";
    }


    /// <summary>
    /// Settings used to recognize business cards
    /// </summary>
    public class BusCardProcessingSettings : IProcessingSettings
    {
        public enum OutputFormatEnum
        {
            vCard, xml, csv
        };

        public BusCardProcessingSettings()
        {
            Language = "English";
            OutputFormat = OutputFormatEnum.vCard;
        }

        /// <summary>
        /// Language used for business card recognition.
        /// You can set several languages like "English,Russian" 
        /// </summary>
        public string Language { get; set; }

        public OutputFormatEnum OutputFormat
        {
            get;
            set; 
        }

        /// <summary>
        /// Any Url parameters that are passed as-is to the server
        /// </summary>
        public String CustomOptions
        {
            get;
            set;
        }

        public string AsUrlParams
        {
            get
            {
                StringBuilder result = new StringBuilder();
                result.AppendFormat("language={0}&exportFormat={1}", Uri.EscapeDataString(Language), OutputFormat.ToString());

                if (!String.IsNullOrEmpty(CustomOptions))
                    result.Append("&" + CustomOptions);

                return result.ToString();
            }
        }
    }
}

File: Java/Abbyy.Ocrsdk.client/src/com/abbyy/ocrsdk/ProcessingSettings.java
package com.abbyy.ocrsdk;

public class ProcessingSettings {

	public String asUrlParams() {
		return String.format("language=%s&exportFormat=%s", language,
				outputFormat);
	}

	public enum OutputFormat {
		txt, rtf, docx, xlsx, pptx, pdfSearchable, pdfTextAndImages, xml
	}

	public void setOutputFormat(OutputFormat format) {
		outputFormat = format;
	}

	public OutputFormat getOutputFormat() {
		return outputFormat;
	}
	
	public String getOutputFileExt() {
		switch( outputFormat ) {
			case txt: return ".txt";
			case rtf: return ".rtf";
			case docx: return ".docx";
			case xlsx: return ".xlsx";
			case pptx: return ".pptx";
			case pdfSearchable:
			case pdfTextAndImages: return ".pdf";
			case xml: return ".xml";
		}
		return ".ocr";
	}

	/*
	 * Set recognition language. You can set any language listed at
	 * http://ocrsdk.com/documentation/specifications/recognition-languages/ or
	 * set comma-separated combination of them.
	 * 
	 * Examples: English English,ChinesePRC English,French,German
	 */
	public void setLanguage(String newLanguage) {
		language = newLanguage;
	}

	public String getLanguage() {
		return language;
	}

	private String language = "English";
	private OutputFormat outputFormat = OutputFormat.pdfSearchable;
}
